src/main/scala/gitbucket/core/controller/RepositoryViewerController.scala

```diff
@@ -526,6 +526,118 @@ trait RepositoryViewerControllerBase extends ControllerBase {
     }
   })
 
+  /**
+    * Upload file to a branch.
+    */
+  post("/:owner/:repository/files/upload")(collaboratorsOnly { repository =>
+    defining(repository.owner, repository.name){ case (owner, name) =>
+      (for {
+        data <- extractFromJsonBody[UploadFiles] if data.isValid
+      } yield {
+        Directory.getAttachedDir(owner, name) match {
+          case dir if (dir.exists && dir.isDirectory) =>
+            val _commitFiles = data.fileIds.map { case (fileName, id) =>
+              dir.listFiles.find(_.getName.startsWith(id + ".")).map { file =>
+                val s = scala.io.Source.fromFile(file) // Codec ????
+              val byteArray = s.map(_.toByte).toArray
```

Comment:
You can use `IOUtils` instead of `scala.io.Source` to get bytes like this:
https://github.com/gitbucket/gitbucket/blob/master/src/main/scala/gitbucket/core/controller/FileUploadController.scala#L66